### DISCUSSION TITLE
WebSocket: How to get Id of WsSession and check if it's open

### DISCUSSION THREAD

**Discussion by dcoracle** [helidon-io/helidon#8508] [Q&A] [answered]
I was trying to upgrade from Helidon 3.x to 4.x and port some WebSocket code but had to halt the effort because WsSession is missing a couple of key methods I used to get from jakarta.websocket.Session.  Namely Session.getId(), and Session.isOpen().  Is there any way to obtain the backing Implementation of WsSession programatically?  Or add these methods to WsSession?

I'm looking at https://helidon.io/docs/latest/apidocs/io.helidon.websocket/io/helidon/websocket/WsSession.html

**Comment by klustria** [comment]
To be clear, are you migrating `jakarta.websocket.Session` to `WsSession` but it is lacking some of the methods that you use in the former?

**Reply by dcoracle** [reply]
correct

**Comment by spericas** [comment, marked as answer]
@dcoracle Helidon 4.x also supports the Jakarta API, as before. Is there a reason for you to migrate to the Helidon WS API?

**Reply by dcoracle** [reply]
Good question :) I think I did a blind port (i.e. changed top level lib from Helidon 3.x to Helidon 4.x in my pom.xml) and it couldn't compile anymore.  I'll have to check my dependencies again.

Thanks
